server.go

```diff
@@ -36,6 +40,17 @@ func (server *Server) Run() {
 	defer listener.Close()
 
 	slog.Info(fmt.Sprintf("tcp server listening on %s:%s", server.host, server.port))
+	conPoolStr := utils.GetEnvVar("CONNECTION_POOL", "1")
```

Comment:
You can use `max.connections` as an env variable name to comply with the KIP. See `https://cwiki.apache.org/confluence/display/KAFKA/KIP-402%3A+Improve+fairness+in+SocketServer+processors` for details, search for `max.connections`. The default value based on the KIP is Int.Max. This essentially means leaving the pool boundless by default with the option for sysadmins to limit connections if needed.

[Here](https://cwiki.apache.org/confluence/display/KAFKA/KIP-402%3A+Improve+fairness+in+SocketServer+processors) is the implementation in Apache Kafka, it implements more functionality, but we  can add only the simple connection pool for now.